example-code/snippets/src/backend/WorkspaceExamples.test.ts

```diff
@@ -0,0 +1,363 @@
+        // A description of the new WorkspaceContainer.
+        metadata: {
+          label: "trees",
+          description: "trees organized by genus",
+        },
+        // Ownership, access control, and datacenter location are defined by the iTwin.
+        scope: { iTwinId },
+        // The name of the default WorkspaceDb to be created inside the new container.
+        dbName: "cornus",
+        // The manifest to be embedded inside the default WorkspaceDb.
+        manifest: {
+          // A user-facing name for the WorkspaceDb.
+          workspaceName: "Dogwood trees",
+          // A description of the WorkspaceDb's contents and purpose.
+          description: "Trees belonging to the genus cornus",
+          // The name of someone (typically an administrator) who can provide help and information
+          // about this WorkspaceDb.
+          contactName: "Lief E. Greene",
+        },
+      });
+      // __PUBLISH_SECTION_END__
+      
+      expect(container.cloudProps).not.to.be.undefined;
+
+      // __PUBLISH_SECTION_START__ WorkspaceExamples.AddTrees
+      interface TreeResource {
+        commonName: string;
+        hardiness: HardinessRange;
+        light: "full" | "shade" | "partial";
+      }
+
+      function addTree(treeDb: EditableWorkspaceDb, species: string, tree: TreeResource): void {
+        treeDb.addString(species, JSON.stringify(tree));
+      }
+
+      container.acquireWriteLock("Lief E. Greene");
+      
+      const cornusDbProps = (await container.createNewWorkspaceDbVersion({
+        fromProps: { dbName: "cornus" },
+        versionType: "minor",
+      })).newDb;
+      
+      let cornusDb = container.getEditableDb(cornusDbProps);
+      cornusDb.open();
+
+      addTree(cornusDb, "alternifolia", {
+        commonName: "Pagoda Dogwood",
+        hardiness: { minimum: 4, maximum: 8 },
+        light: "full",
+      });
+
+      addTree(cornusDb, "asperifolia", {
+        commonName: "Roughleaf Dogwood",
+        hardiness: { minimum: 9, maximum: 9 },
+        light: "full",
+      });
+
+      cornusDb.close();
+      container.releaseWriteLock();
+      // __PUBLISH_SECTION_END__
+      expect(cornusDb.cloudProps).not.to.be.undefined;
+      expect(cornusDb.cloudProps!.version).to.equal("1.1.0");
+      
+      // __PUBLISH_SECTION_START__ WorkspaceExamples.CreatePatch
+      container.acquireWriteLock("Lief E. Greene");
+      const cornusPatchProps = (await container.createNewWorkspaceDbVersion({
+        fromProps: { dbName: "cornus", version: "1.1.0" },
+        versionType: "patch",
+      })).newDb;
+      
+      cornusDb = container.getEditableDb(cornusPatchProps);
+      cornusDb.open();
+      addTree(cornusDb, "racemosa", {
+        commonName: "Northern Swamp Dogwood",
+        hardiness: { minimum: 4, maximum: 9 },
+        light: "full",
+      });
+      cornusDb.close();
+
+      await container.createDb({
```

Comment:
@kabentley `WorkspaceEditor.createDb` appears to be unusable. It will upload a new, empty db, and then throw an error trying to retrieve it for editing.

```
Error: abies:1.0.0 has been published and is not editable. Make a new version first.
      at EditorContainerImpl.getEditableDb (/home/paul/c/j2/s/itwinjs/core/backend/src/internal/workspace/WorkspaceImpl.ts:553:13)
      at EditorContainerImpl.getWorkspaceDb (/home/paul/c/j2/s/itwinjs/core/backend/src/internal/workspace/WorkspaceImpl.ts:548:17)
      at EditorContainerImpl.createDb (/home/paul/c/j2/s/itwinjs/core/backend/src/internal/workspace/WorkspaceImpl.ts:591:17)
      at Context.<anonymous> (src/backend/WorkspaceExamples.test.ts:327:7)
```

Reply:
Working around it by making 1 db per container (the default db), which is the pattern I want to encourage anyway, but we shuold `createDb`.

Reply:
As we discussed, `getEditableDb` should take an optional parameter that skips the "copied" check for amending previously published WorkspaceDbs (the opposite of the NPM paradigm.) Then, `createDb` would pass `true` for that.

Reply:
Done.